application/src/main/java/org/togetherjava/tjbot/commands/moderation/KickCommand.java

```diff
@@ -0,0 +1,118 @@
+package org.togetherjava.tjbot.commands.moderation;
+
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+import java.util.Objects;
+
+
+/**
+ * This command can kicks users. Kicking can also be paired with a kick reason. The command will
+ * also try to DM the user to inform them about the action and the reason.
+ * <p>
+ * The command fails if the user triggering it is lacking permissions to either kick other users or
+ * to kick the specific given user (for example a moderator attempting to kick an admin).
+ *
+ */
+public final class KickCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(KickCommand.class);
+    private static final String USER_OPTION = "user";
+    private static final String REASON_OPTION = "reason";
+
+    /**
+     * Constructs an instance
+     *
+     * @see KickCommand
+     */
+    public KickCommand() {
+        super("kick", "Kicks the given user from the server", SlashCommandVisibility.GUILD);
+
+        getData().addOption(OptionType.USER, USER_OPTION, "The user who you want to kick", true)
+            .addOption(OptionType.STRING, REASON_OPTION, "Why the user should be kicked", true);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandEvent event) {
+        OptionMapping userOption =
+                Objects.requireNonNull(event.getOption(USER_OPTION), "The target is null");
+
+        Member targetMember = userOption.getAsMember();
```

Comment:
And I'd still check or the target is in this guild, if it's not in this guild it'd throw an error and that looks ugly for the logging

Reply:
No need since userOption checks

Reply:
How does it?
It accepts any user, so even that're not in this guild.

Reply:
no it gets the member then if the member is not null it gets it as a user